lua/jdtls/dap.lua

```diff
@@ -324,15 +324,12 @@ local function make_config(lens, launch_args, config_overrides)
   }
   config = vim.tbl_extend('force', config, config_overrides or default_config_overrides)
   if lens.testKind == TestKind.TestNG or lens.kind == TestKind.TestNG then
-    config.mainClass = 'org.testng.TestNG'
+    config.mainClass = 'com.microsoft.java.test.runner.Launcher'
     -- id is in the format <project>@<class>#<method>
     local parts = vim.split(lens.id, '@')
     parts  = vim.split(parts[2], '#')
-    if #parts > 1 then
-      config.args = string.format('-testclass %s -methods %s.%s', parts[1], parts[1], parts[2])
-    else
-      config.args = string.format('-testclass %s', parts[1])
-    end
+    config.args = string.format('testng %s', lens.fullName)
+    table.insert(config.classPaths, os.getenv 'HOME' .. '/.local/share/nvim/mason/packages/java-test/extension/server/com.microsoft.java.test.runner-jar-with-dependencies.jar');
```

Comment:
Hardcoding paths / depending on mason is not an option.

Is this even needed?

Reply:
Otherwise 'com.microsoft.java.test.runner.Launcher' isn't found. Maybe adding init_options.bundles to the classPaths instead?

Reply:
The runner itself might actually not be part of the `bundles` but it could be an option to search the base dirs of the bundles for the jar and add it if found. And if it isn't either throw an error or fallback to the old logic. (I'd prefer the fallback to not break anything for users)

Reply:
Sounds good. I've added a check, if the runner is in the bundles. If not, it falls back to the previous config and prints a message.